### PR TITLE
Routing: Try router5

### PR DESCRIPTION
**Pull request by ehg**
Try a more declarative router that doesn't use React components to specify routes, is isomorphic and isn't page.js.

_WIP_

### CODE REVIEW

File: client/create-router.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	return router;
+}
+
+export function ensureMiddleware( /* router */ ) {
+	return ( toState, fromState, done ) => {
+		const isSection = sectionName => sections.get().filter( s => s.name === sectionName );
+		console.log( 'toState', toState, 'fromState', fromState, 'done', done );
+		if ( isSection( toState.name ) ) {
+			console.log( 'preloading', toState.name );
+			sections.preload( toState.name );
```
Comment (mcsf) [inline client/create-router.js]: Nice.

File: client/my-sites/themes/controller.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ var SingleSiteComponent = require( 'my-sites/themes/single-site' ),
 var controller = {
 
 	themes: function( context ) {
+		context = context || {};
```
Comment (mcsf) [inline client/my-sites/themes/controller.js]: Why not `function( context = {} ) {`?

File: client/state/index.js [hunk 3]
```diff
@@ -48,16 +50,9 @@ if ( typeof window === 'object' ) {
...
-		typeof window === 'object' &&
-		window.app &&
-		window.app.isDebug &&
-		window.devToolsExtension
-	) {
-		createStoreWithMiddleware = compose( createStoreWithMiddleware, window.devToolsExtension() );
-	}
+export function createReduxStore( initialState = {}, router ) {
+	console.log( router );
+	middleware = [ router5Middleware( router ), ...middleware ];
```
Comment (mcsf) [inline client/state/index.js]: If this function somehow gets called more than once, `middleware` will keep growing incorrectly.

**Comment by ehg** [comment]
Closing this in favour of #3065, but leaving the branch in case it's useful later.

**Pull request closed without merging**